### PR TITLE
Force 3 significant figures in Y axis values

### PR DESCRIPTION
**Pull request by drewgillies**
Use adjusted rounding and formatting to get the required significant figures. Added logic to append trailing zeroes for visual alignment.

Addresses https://github.com/NuGet/NuGetGallery/issues/7972

image

### CODE REVIEW

File: src/NuGetGallery/ViewModels/StatisticsPackagesViewModel.cs [hunk 1]
```diff
@@ -142,22 +142,37 @@ public string DisplayPercentage(float amount, float total)
             return (amount / total).ToString("P0", CultureInfo.CurrentCulture);
         }
 
-        public string DisplayShortNumber(long number)
+        public string DisplayShortNumber(double number, int sigFigures = 3)
```
Comment (loic-sharma) [inline src/NuGetGallery/ViewModels/StatisticsPackagesViewModel.cs:145]: Can we make this method static and add unit tests for it?

File: src/NuGetGallery/ViewModels/StatisticsPackagesViewModel.cs [hunk 2]
```diff
@@ -142,22 +142,37 @@ public string DisplayPercentage(float amount, float total)
...
 
             while (number >= 1000)
             {
-                number = number / 1000;
+                number /= 1000;
                 numDiv++;
             }
 
-            if (numDiv >= _magnitudeAbbreviations.Length)
+            // Find a rounding factor based on size, and round to sigFigures, e.g. for 3 sig figs, 1.774545 becomes 1.77.
```
Comment (skofman1) [inline src/NuGetGallery/ViewModels/StatisticsPackagesViewModel.cs:155]: really hard to review without UTs..

Comment (drewgillies) [reply src/NuGetGallery/ViewModels/StatisticsPackagesViewModel.cs]: Sorry about that--UT added with a bunch of cases.

File: src/NuGetGallery/ViewModels/StatisticsPackagesViewModel.cs (hunk 1 above)
Comment (drewgillies) [reply src/NuGetGallery/ViewModels/StatisticsPackagesViewModel.cs]: Done

File: tests/NuGetGallery.Facts/ViewModels/StatisticsPackagesViewModelFacts.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        [InlineData(1775856283, 5, "1.7759B")]
+        [InlineData(775856283, 3, "776M")]
+        [InlineData(775856283, 5, "775.86M")]
+        [InlineData(75856283, 3, "75.9M")]
+        [InlineData(56283, 3, "56.3k")]
+        [InlineData(56283283283283, 3, "56.3T")]
+        [InlineData(56283283283283283, 3, "56.3q")]
+        [InlineData(56283283283283283283283283283283d, 3, "56.3n")]
+        [InlineData(56283283283283283283283283283283283d, 3, "56.3 10^33")]
+        public void CreatesShortNumberRespectingSignificantFigures(double number, int sigFigs, string expected)
```
Comment (loic-sharma) [inline tests/NuGetGallery.Facts/ViewModels/StatisticsPackagesViewModelFacts.cs:25]: Could you also add some tests for already short numbers like `1` or `100`?

Comment (drewgillies) [reply tests/NuGetGallery.Facts/ViewModels/StatisticsPackagesViewModelFacts.cs:28]: Good idea--thanks. Done.

**Pull request merged**